service/migration_manager.cc

```diff
@@ -222,7 +222,7 @@ void migration_manager::schedule_schema_pull(const gms::inet_address& endpoint,
 }
 
 bool migration_manager::have_schema_agreement() {
-    const auto known_endpoints = _gossiper.get_endpoint_states();
+    const auto known_endpoints = _gossiper.get_live_endpoint_states();
```

Comment:
True.  It's not a problem, but I can document that in the change log.
In a follow up series to change the gossiper interface to support host_id also `get_endpoint_states` is going to return a copy of the endpoint states, although in the future model the endpoint_state is going to be kept in a lw_shared_ptr so the copy of the map can still point to the shared state, reducing the cost.

Reply:
This is a unnecessary change for this big enough series. If you want to avoid eps.is_alive() just call g.is_alive(). 

I do not think it worth adding such expensive function.

Reply:
I think that a `get_live_endpoint_states` can be more efficient than calling g.is_alive for every endpoint since it can simply walk `_live_endpoints` and get (a lw_shared_ptr of) the endpoint state for each endpoint, but w ecan indeed leave it to a follow-up for brevity sake.

Reply:
The have_schema_agreement is used only in two places. Both of them are in slow path. I'd suggest drop get_live_endpoint_states for now.